storage/Devicegraph.cc

```diff
@@ -43,13 +44,15 @@ namespace storage
 
 
     WrongNumberOfParents::WrongNumberOfParents(size_t seen, size_t expected)
-	: Exception(sformat("wrong number of parents, seen '%zu', expected '%zu'", seen, expected))
+	: Exception(sformat("wrong number of parents, seen '%zu', expected '%zu'", seen, expected),
+		    Silencer::is_any_active() ? DEBUG : WARNING)
```

Comment:
now all over place is distributed that we silence from WARN to DEBUG. If we want to silence to given level, then it make sense for me to include target level for silence in silencer and not all over code. So change will be only that it say log level it want initially use and silencer decide if it want to silence it to silence level. I do not see how it can be worse then now.
instead of `Silencer::is_any_active() ? DEBUG : WARNING` will be `Silencer::silence(WARNING)` which return WARNING or DEBUG depending if silencer is on or off